python/triton/tools/cppjit.py

```diff
@@ -0,0 +1,73 @@
+import importlib.util
+import sys
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import List
+
+import triton
+import triton.language as tl
+
+desc = """
+Triton compiler for C++ JIT.
+
+This program compiles the kernel with name `kernel-name` in the file at the
+provided `path` into self-contained C++ source-code that embeds the MLIR code.
+
+signature is provided as a list of (optionally divisibility-hinted) types
+or constexpr values, e.g.
+
+`cppjit.py --kernel-name kernel --signature "*f32, i32, i32, i32" --out-name kernel /path/to/kernel.py`
+
+will compile triton.JITFunction of name `kernel` inside the file `/path/to/kernel.py`.
+Said kernel will be specialized such that argument 0 is f32 buffer, argument 1, 2, 3 are i32 scalars.
+Note that, the `tl.constexpr` arguments should be specialized too.
+"""
+
+if __name__ == "__main__":
+
+    # command-line arguments
+    parser = ArgumentParser(description=desc)
+    parser.add_argument("path", help="Path to Python source containing desired kernel in its scope. File will be executed.")
+    parser.add_argument("--kernel-name", "-n", type=str, default="", help="Name of the kernel to compile", required=True)
+    parser.add_argument("--out-name", "-on", type=str, default=None, help="Out name for the compiled kernel")
+    parser.add_argument("--out-path", "-o", type=Path, default=None, help="Out filename")
+    parser.add_argument("--signature", "-s", type=str, help="Signature of the kernel", required=True)
```

Comment:
`cppjit.py` toolkit has a similar user interface with `compile.py`, it accepts signature and output a TritonLang mlir module (and the other launcher stuff later).

Note that, unlike in `compile.py`, the constexpr arguments require some constant int, here for `cppjit.py`, the data types are required instead. For instance, a `BLOCK_M` constexpr should be specified with an `i32:c`, `c` is just a mark for constexpr.

 We believe that the constexpr arguments should be static type for simple tiling settings, so it is reasonable to (partially) specify their types in cppjit.py